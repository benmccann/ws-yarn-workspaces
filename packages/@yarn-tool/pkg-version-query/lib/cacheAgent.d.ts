import Cache from 'lru-cache-fs2';
export declare function initCache(): Cache<string, import("ts-type").ITSRequireAtLeastOne<import("./types").ICachedVersionResultCore, "result" | "error">>;
export declare function getCache(): Cache<string, import("ts-type").ITSRequireAtLeastOne<import("./types").ICachedVersionResultCore, "result" | "error">>;
