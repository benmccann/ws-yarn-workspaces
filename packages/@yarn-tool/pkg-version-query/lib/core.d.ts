import { Options, AbbreviatedVersion, AbbreviatedMetadata, FullMetadataOptions, FullMetadata } from 'package-json';
import Bluebird from 'bluebird';
import { IOptionsQueryVersion } from './types';
export declare function _queryVersion(packageName: string, options: IOptionsQueryVersion<FullMetadataOptions>): Bluebird<FullMetadata>;
export declare function _queryVersion(packageName: string, options: Omit<Options, 'allVersions' | 'version'> & {
    allVersions?: false;
    version: string;
}): Bluebird<AbbreviatedVersion>;
export declare function _queryVersion(packageName: string, options?: IOptionsQueryVersion<Options>): Bluebird<AbbreviatedMetadata>;
