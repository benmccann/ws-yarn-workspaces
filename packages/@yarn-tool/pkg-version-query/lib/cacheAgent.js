"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.initCache = initCache;
exports.getCache = getCache;
const tslib_1 = require("tslib");
const lru_cache_fs2_1 = tslib_1.__importDefault(require("lru-cache-fs2"));
const cache_path_1 = require("cache-path");
let cache;
function initCache(options) {
    let id = '@yarn-tool/pkg-version-query';
    let cwd = (0, cache_path_1.getCachePath)(id, {
        fnOrder: [
            cache_path_1.findYarnCachePath,
            cache_path_1.findNpmCachePath,
            cache_path_1.findOSTempPath,
            cache_path_1.findPkgModulePath,
        ],
    });
    let { cacheAgentOptions } = (options !== null && options !== void 0 ? options : (options = {}));
    cacheAgentOptions = {
        //		max: 1000,
        //		maxAge: 5 * 60 * 1000,
        ...cacheAgentOptions,
        cacheName: (0, cache_path_1.normalizeName)(id, true),
        autoCreate: true,
        cwd,
    };
    if (!cacheAgentOptions.max || cacheAgentOptions.max <= 100) {
        cacheAgentOptions.max = 1000;
    }
    if (!cacheAgentOptions.maxAge || cacheAgentOptions.maxAge <= 60 * 1000) {
        cacheAgentOptions.maxAge = 10 * 60 * 1000;
    }
    const cache = new lru_cache_fs2_1.default(cacheAgentOptions);
    return cache;
}
function getCache(options) {
    var _a;
    return (_a = options === null || options === void 0 ? void 0 : options.cacheAgent) !== null && _a !== void 0 ? _a : (cache !== null && cache !== void 0 ? cache : (cache = initCache(options)));
}
//# sourceMappingURL=cacheAgent.js.map