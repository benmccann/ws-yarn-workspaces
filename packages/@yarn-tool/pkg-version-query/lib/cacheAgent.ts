import Cache from 'lru-cache-fs2';
import {
	normalizeName,
	getCachePath,
	findPkgModulePath,
	findYarnCachePath,
	findNpmCachePath,
	findOSTempPath,
} from 'cache-path';
import { ICachedVersionResult } from './types';

let cache: Cache<string, ICachedVersionResult>;

export function initCache()
{
	let id = '@yarn-tool/pkg-version-query';

	let cwd = getCachePath(id, {
		fnOrder: [
			findYarnCachePath,
			findNpmCachePath,
			findOSTempPath,
			findPkgModulePath,
		],
	})

	const cache = new Cache<string, ICachedVersionResult>({
		max: 100,
		cacheName: normalizeName(id, true),
		cwd,
		autoCreate: true,
		maxAge: 60 * 1000,
	});

	return cache;
}

export function getCache()
{
	return cache ??= initCache();
}

