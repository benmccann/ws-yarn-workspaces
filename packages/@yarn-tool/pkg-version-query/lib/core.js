"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports._queryVersion = void 0;
const tslib_1 = require("tslib");
const package_json_1 = (0, tslib_1.__importDefault)(require("package-json"));
const bluebird_1 = (0, tslib_1.__importDefault)(require("bluebird"));
function _queryVersion(packageName, options) {
    options !== null && options !== void 0 ? options : (options = {});
    return bluebird_1.default.resolve((0, package_json_1.default)(packageName, options))
        .catch(e => {
        if (options.notThrowError) {
            return;
        }
        return Promise.reject(e);
    });
}
exports._queryVersion = _queryVersion;
//# sourceMappingURL=core.js.map