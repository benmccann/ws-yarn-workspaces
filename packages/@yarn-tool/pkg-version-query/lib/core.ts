import _packageJson, {
	Options,
	AbbreviatedVersion,
	AbbreviatedMetadata,
	FullMetadataOptions,
	FullMetadata,
} from 'package-json';
import Bluebird from 'bluebird';
import { IOptionsQueryVersion } from './types';

export function _queryVersion(packageName: string,
	options: IOptionsQueryVersion<FullMetadataOptions>,
): Bluebird<FullMetadata>;
export function _queryVersion(packageName: string,
	options: Omit<Options, 'allVersions' | 'version'> & {
		allVersions?: false,
		version: string,
	},
): Bluebird<AbbreviatedVersion>
export function _queryVersion(packageName: string,
	options?: IOptionsQueryVersion<Options>,
): Bluebird<AbbreviatedMetadata>
export function _queryVersion(packageName: string, options?: IOptionsQueryVersion<Options>)
{
	options ??= {};

	return Bluebird.resolve(_packageJson(packageName, options))
		.catch(e =>
		{
			if (options.notThrowError)
			{
				return
			}

			return Promise.reject(e)
		}) as any
		;
}
