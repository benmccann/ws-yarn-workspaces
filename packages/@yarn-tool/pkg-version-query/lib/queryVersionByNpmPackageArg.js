"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.parseVersionByNpmPackageArg = parseVersionByNpmPackageArg;
exports._parseVersionByNpmPackageArgCore = _parseVersionByNpmPackageArgCore;
exports.queryVersionByNpmPackageArgWithCache = queryVersionByNpmPackageArgWithCache;
const npm_package_arg_util_1 = require("@yarn-tool/npm-package-arg-util");
const queryVersion_1 = require("./queryVersion");
function parseVersionByNpmPackageArg(input) {
    return _parseVersionByNpmPackageArgCore((0, npm_package_arg_util_1.npa)(input));
}
function _parseVersionByNpmPackageArgCore(result) {
    const { name, fetchSpec: version } = result;
    return {
        name,
        version,
    };
}
function queryVersionByNpmPackageArgWithCache(input, options) {
    const { name, version } = parseVersionByNpmPackageArg(input);
    return (0, queryVersion_1.queryVersionWithCache)(name, version, options);
}
exports.default = queryVersionByNpmPackageArgWithCache;
//# sourceMappingURL=queryVersionByNpmPackageArg.js.map