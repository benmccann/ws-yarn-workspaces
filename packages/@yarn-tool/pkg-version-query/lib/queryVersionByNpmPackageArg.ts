import npmPackageArg from 'npm-package-arg';
import queryVersionWithCache from './queryVersion';
import { IOptionsQueryVersion } from './types';
import { Options } from 'package-json';

export function parseVersionByNpmPackageArg(input: string): {
	name: string;
	version: string;
}
{
	const { name, fetchSpec: version } = npmPackageArg(input);

	return {
		name,
		version,
	}
}

export function queryVersionByNpmPackageArgWithCache(input: string, options?: IOptionsQueryVersion<Options>)
{
	const { name, version } = parseVersionByNpmPackageArg(input);

	return queryVersionWithCache(name, version, options)
}

export default queryVersionByNpmPackageArgWithCache
