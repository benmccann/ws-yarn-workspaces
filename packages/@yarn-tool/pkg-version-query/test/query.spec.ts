import { parseVersionByNpmPackageArg } from '../lib/queryVersionByNpmPackageArg';

describe(`parseVersionByNpmPackageArg`, () =>
{

	test(`typescript@>= 1 & < 2`, () =>
	{

		expect(parseVersionByNpmPackageArg(`typescript@>= 1 & < 2`)).toMatchSnapshot();

	});

	test(`typescript@1`, () =>
	{

		expect(parseVersionByNpmPackageArg(`typescript@1`)).toMatchSnapshot();

	});

	test(`typescript`, () =>
	{

		expect(parseVersionByNpmPackageArg(`typescript`)).toMatchSnapshot();

	});

})
