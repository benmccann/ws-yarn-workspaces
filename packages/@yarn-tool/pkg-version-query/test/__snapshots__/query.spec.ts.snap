// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseVersionByNpmPackageArg typescript 1`] = `
{
  "name": "typescript",
  "version": "*",
}
`;

exports[`parseVersionByNpmPackageArg typescript@>= 1 & < 2 1`] = `
{
  "name": "typescript",
  "version": ">= 1 & < 2",
}
`;

exports[`parseVersionByNpmPackageArg typescript@1 1`] = `
{
  "name": "typescript",
  "version": "1",
}
`;
